package.json

```diff
@@ -2,48 +2,66 @@
-    "@angular-eslint/eslint-plugin-template": "^15.2.1",
-    "@angular-eslint/template-parser": "^15.2.1",
-    "@angular/cli": "^15.2.4",
-    "@angular/compiler-cli": "^15.2.4",
-    "@angular/language-service": "^15.2.4",
-    "@commitlint/cli": "^17.5.1",
-    "@commitlint/config-conventional": "^17.4.4",
-    "@nrwl/angular": "15.8.9",
-    "@nrwl/eslint-plugin-nx": "15.8.9",
-    "@nrwl/jest": "15.8.9",
-    "@nrwl/linter": "15.8.9",
-    "@nrwl/storybook": "15.8.9",
-    "@nxext/stencil": "^15.7.0",
-    "@stencil/angular-output-target": "^0.4.0",
-    "@stencil/core": "^2.22.3",
-    "@stencil/sass": "^2.0.4",
-    "@types/jest": "28.1.1",
+    "@angular-builders/custom-webpack": "^17",
+    "@angular/cli": "^17.3.2",
+    "@angular/compiler-cli": "^17",
+    "@angular/language-service": "^17",
+    "@angular/localize": "^17",
+    "@commitlint/cli": "^19.2.1",
+    "@commitlint/config-conventional": "^19.1.0",
+    "@nx/angular": "18.1.3",
+    "@nx/eslint": "18.1.3",
+    "@nx/eslint-plugin": "18.1.3",
+    "@nx/jest": "18.1.3",
+    "@nx/storybook": "18.1.3",
+    "@nxext/stencil": "^18",
+    "@stencil/angular-output-target": "^0.8.4",
+    "@stencil/core": "^4.13.0",
+    "@stencil/sass": "^3.0.11",
+    "@stencil/webpack": "^0.0.6",
+    "@types/dom-mediacapture-record": "^1.0.15",
+    "@types/emscripten": "^1.39.6",
+    "@types/file-saver": "^2.0.5",
+    "@types/jasmine": "^5.1.4",
     "@types/node": "^16.18.22",
-    "@typescript-eslint/eslint-plugin": "^5.57.0",
-    "@typescript-eslint/parser": "^5.57.0",
-    "eslint": "~8.15.0",
-    "eslint-config-prettier": "8.1.0",
-    "husky": "^8.0.3",
-    "jest": "28.1.1",
-    "jest-environment-jsdom": "28.1.1",
-    "jest-preset-angular": "~12.2.3",
-    "lerna": "^5.6.2",
-    "ng-packagr": "^15.2.2",
-    "nx": "15.8.9",
-    "postcss-preset-env": "^7.8.3",
-    "prettier": "^2.8.7",
-    "pretty-quick": "^3.1.3",
-    "ts-jest": "28.0.5",
-    "ts-node": "10.9.1",
-    "typescript": "^4.9.5"
+    "@typescript-eslint/eslint-plugin": "^6.13.0",
+    "@typescript-eslint/parser": "^6.13.0",
+    "find-cache-dir": "^3.3.2",
+    "husky": "^9.0.11",
+    "jasmine-core": "^5.1.2",
+    "jest": "^29",
+    "jest-environment-jsdom": "^29",
+    "karma": "^6.4.3",
+    "karma-chrome-launcher": "^3.2.0",
+    "karma-coverage": "^2.2.1",
+    "karma-jasmine": "^5.1.0",
+    "karma-jasmine-html-reporter": "^2.1.0",
+    "ng-packagr": "^17",
+    "nx": "18.1.3",
+    "prettier": "^3.2.5",
+    "pretty-quick": "^4.0.0",
+    "ts-jest": "^29"
+  },
+  "dependencies": {
+    "@angular/animations": "^17",
+    "@angular/cdk": "^17",
+    "@angular/common": "^17",
+    "@angular/compiler": "^17",
+    "@angular/core": "^17",
+    "@angular/forms": "^17",
+    "@angular/material": "^17",
+    "@angular/platform-browser": "^17",
+    "@angular/router": "^17",
+    "bootstrap": "^5.2.3",
+    "file-saver": "^2.0.5",
+    "image-conversion": "^2.1.1",
+    "ngx-toastr": "^16.1.0",
```

Comment:
I've seen this with nx, where all the dependencies end up in the main `package.json` but I don't understand it. How does it know to not bundle `ngx-toastr` with the web-component for example?

Reply:
the bundler will only bundle stuff that the code actually uses - but als note that the web-component declares the dependencies used by its actual code in `packages/web-component/package.json`